### PR TITLE
Climber

### PR DESCRIPTION
**Pull request by ProgrammingSR**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -68,7 +68,7 @@ dependencies {
     simulationRelease wpi.sim.enableRelease()
 
     testImplementation 'org.junit.jupiter:junit-jupiter:5.10.1'
-    testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
```
Comment (ReeledWarrior14) [inline build.gradle]: What is this?

File: simgui-ds.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
+{
```
Comment (ReeledWarrior14) [inline simgui-ds.json]: Remove all of these changes, or see what the difference is after merging main into this branch properly.

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        public static final int kRearLeftTurningMotorPort = 22;
+        public static final int kFrontRightTurningMotorPort = 37;
+        public static final int kRearRightTurningMotorPort = 26;
+
+        public static final int kFrontLeftTurningEncoderPort = 19;
+        public static final int kRearLeftTurningEncoderPort = 20;
+        public static final int kFrontRightTurningEncoderPort = 18;
+        public static final int kRearRightTurningEncoderPort = 17;
+
+        public static final double kFrontLeftTurningEncoderOffset = 0;
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/Constants.java]: These were removed in main, they should not be here.

File: src/main/java/frc/robot/Constants.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        /** Diameter of each wheel in the SDS MK4i swerve module (in meters) */
+        public static final double kWheelDiameterMeters = 0.1;
+
+        /** Gear ratio between the motor and the wheel. */
+        public static final double kDrivingGearRatio = 8.14; // SDS MK4i's in L1 Configuration
+        // public static final double kDrivingGearRatio = 6.75; // SDS MK4i's in L2
+        // configuration
+
+        // TODO: Tune this PID before running on a robot on the ground
+        public static final double kPModuleTurningController = -0.3;
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/Constants.java]: Again, a change made in main, not updated/being redone in this branch.

File: src/main/java/frc/robot/Constants.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        // ^^ Calculated using the method taken from the old SDS github example
+
+        /** Heading Correction */
+        public static final double kHeadingCorrectionTurningStopTime = 0.2;
+        // TODO: Tune this PID before running on a robot on the ground
+        public static final double kPHeadingCorrectionController = 5;
+    }
+
+    // Intake PID and Encoder Constants
+    public static class IntakeConstants {
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/Constants.java]: Intake and shooter changes should not be on this branch.

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -4,17 +4,99 @@
...
+ * (including subsystems, commands, and button mappings) should be declared here.
+ */
 public class RobotContainer {
+  // The robot's subsystems and commands are defined here
+  private final DriveSubsystem m_robotDrive = new DriveSubsystem();
+  public final ShooterSubsystem m_shooterSubsystem = new ShooterSubsystem();
+  private final ClimberSubsystem m_climberSubsystem = new ClimberSubsystem();
+
+  private final XboxController m_driverController = new XboxController(IOConstants.kDriverControllerPort);
+  private final XboxController m_operatorController = new XboxController(IOConstants.kDriverControllerPort);
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/RobotContainer.java]: Operator controller should not be initialized with the driver controller port. This is a major bug.

File: src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.Compressor;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/subsystems/ClimberSubsystem.java]: Unused import, please remove.

File: src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    public ClimberSubsystem(){
+        m_leftSolenoid.set(kOff);
+        m_rightSolenoid.set(kOff);
+        m_compressor.disable();
+        m_compressor.enableAnalog(ClimberConstants.minPressure, ClimberConstants.maxPressure);
+    }
+    
+    //Runs once every tick (~20ms)
+    public void periodic() {
+        // SmartDashboard.putNumber("Pressure", m_compressor.getPressure());
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/subsystems/ClimberSubsystem.java]: These should not be commented out or they should be removed.

File: src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    //TODO: fix doc
+    /*
+     * Extends both arms
+     */
+    public void forward() {
+        m_leftSolenoid.set(kOff);
+        m_rightSolenoid.set(kOff);
+    }
+
+    //TODO:fix doc
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/subsystems/ClimberSubsystem.java]: This is just... its literally a one line java doc, it takes more effort to write the TODO to fix it than to actually just fix it if there is something wrong with it.

File: src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    public void reverse() {
+        m_leftSolenoid.set(kOff);
+        m_rightSolenoid.set(kOff);
+    }
+    /*
+     * Toggles the state of the climber
+     */
+
+    public void toggle() {
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/subsystems/ClimberSubsystem.java:62]: Formatting is wrong here and none of the Javadocs are done properly.

File: src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        m_leftSolenoid.set(kOff);
+        m_rightSolenoid.set(kOff);
+    }
+
+    //TODO: fix doc
+    /*
+     * Extends both arms
+     */
+    public void forward() {
+        m_leftSolenoid.set(kOff);
```
Comment (ReeledWarrior14) [inline src/main/java/frc/robot/subsystems/ClimberSubsystem.java]: Literally all of these methods with different names set the solenoids to kOff. This is another major bug that really should never even have been committed.

**Review by ReeledWarrior14 — DISMISSED** [review]
This does not look like it will work. I think we will need to wait until it can be tested before merging it. There are a lot of issues that need to be fixed. Also please update the branch before saying it is ready for a review, half the changes make no sense since it is not up to date with main.

**Review by dpk2442 — APPROVED** [review]

**Pull request merged**